lib/binary_search_tree.ex

```diff
@@ -28,6 +28,20 @@ defmodule BinarySearchTree do
     end
   end
 
+  def balance(node) when is_map(node), do: balance(in_order(node))
+  def balance([]), do: nil
+  def balance(list) do
+    mid = div(length(list), 2)
+    {left, right} = Enum.split(list, mid)
+
+    cond do
```

Comment:
I'd like to make this a bit more elegant. I was trying to do [node|tail] on an empty list, which blow sup.